src/main/java/com/example/project/repository/book/BookRepository.java

```diff
@@ -1,9 +1,10 @@
 package com.example.project.repository.book;
 
 import com.example.project.model.Book;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
-
+    List<Book> findAllByCategoriesId(Long categoryId); // added
```

Comment:
please remove such comments, they are unnecessary

```suggestion
    List<Book> findAllByCategoriesId(Long categoryId);
```

Reply:
sure